File: src/com/brianc/graphics/SolidColor.java
package com.brianc.graphics;

import java.awt.Graphics2D;

import com.brianc.css.Color;
import com.brianc.layout.Rect;

public class SolidColor implements DisplayCommand {
	final Color color;
	final Rect rect;

	public SolidColor(Color color, Rect rect) {
		this.color = color;
		this.rect = rect;
	}	
	
	@Override
	public void paint(Graphics2D g) {
		g.setColor(new java.awt.Color(color.r, color.g, color.b, color.a));
		g.fill(rect);
	}
}

File: src/com/brianc/graphics/SolidColor.kt
package com.brianc.graphics

import java.awt.Graphics2D

import com.brianc.css.Color
import com.brianc.layout.Rect

class SolidColor(internal val color: Color, internal val rect: Rect) : DisplayCommand {

    override fun paint(g: Graphics2D) {
        g.color = java.awt.Color(color.r.toInt(), color.g.toInt(), color.b.toInt(), color.a.toInt())
        g.fill(rect)
    }
}
